### PR TITLE
Add hiera-eyaml as secrets' encryption optional solution

### PR DESCRIPTION
**Pull request by mkubaczyk**
While helm-secrets (based on sops) seems to be a good option to start with Helmsman, it has proven to be not well suited to multi-person work bringing a headache whenever more than one person tried to edit the same secrets file. It's nature of having metadata, MAC etc. inside of a file makes it really hard to resolve conflicts and merge two changes at once. That is why I'm proposing hiera-eyaml, which we have battle-tested in company for years in Puppet, to be the optional backend for yaml secrets. It allows specific lines encryption, leaving other values unencrypted. It has no state data in file, so the values can be merged like any other changes in git.
One can start using it easily, splitting values and secrets into two files (as it was for now), or just start using secrets files without value files and keep all data in one yaml, where some values are encrypted and some are not.

### CODE REVIEW

**Comment by mkubaczyk** [comment]
@luisdavim how would I run tests with additional dependency needed for tests? :-)
I need to have gem hiera-eyaml installed (added to test_files/dockerfile), but can't pass tests since current docker image used for tests run doesn't have it

**Comment by sami-alajrami** [comment]
@mkubaczyk the test image is currently manually updated and pushed. I have built it with your new deps and pushed it. 
We will find a way to automate building and pushing that image before the source build and test are run.

**Comment by mkubaczyk** [comment]
@sami-alajrami thanks. Tests are fine now.
So at this point I'd like to see what you guys think about my changes. :-)

File: docs/how_to/settings/use-hiera-eyaml-as-secrets-encryption.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+version: v1.13.0
+---
+
+# Using hiera-eyaml as backend for secrets' encryption
+
+Helmsman uses helm-secrets as a default solution for secrets' encryption. 
+And while it is a good off-the-shelve solution it may quickly start causing problems when few developers starts working on the secrets files simultaneously.
+SOPS-based secrets can be easily merged or rebased in case of conflicts etc.
```
Comment (krogon) [inline docs/how_to/settings/use-hiera-eyaml-as-secrets-encryption.md]: I think you meant `can not`

**Comment by luisdavim** [comment]
@mkubaczyk @sami-alajrami I think once we merge this we should probably cut a new release, we've had a lot of changes merged in since the last release.

**Comment by sami-alajrami** [comment]
Yes, please go ahead with a new release.

**Comment by mkubaczyk** [comment]
I assumed this changes go as 1.13.0 so this release was marked in docs of features of mine

**Comment by luisdavim** [comment]
> I assumed these changes go as 1.13.0 so this release was marked in docs of features of mine

Yeah, we can cut `v1.13.0` once this is merged.

**Comment by mkubaczyk** [comment]
I'd like @sami-alajrami to also review this PR and once approved, let's go with 1.13.0

**Comment by mkubaczyk** [comment]
Ok, I don't want to block release of 1.13.0, so since I have approve from @luisdavim, I'll merge it

**Pull request merged**